processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -173,7 +220,7 @@ public Filter toFilter()
   @Override
   public RangeSet<String> getDimensionRangeSet(String dimension)
   {
-    if (!Objects.equals(getDimension(), dimension) || getExtractionFn() != null || alphaNumeric) {
+    if (!Objects.equals(getDimension(), dimension) || getExtractionFn() != null || !ordering.equals(StringComparatorType.LEXICOGRAPHIC)) {
```

Comment:
hmm, this would probably be easier to read as a NOT of ANDs rather than an OR of NOTs.